ch_pipeline/processing/base.py

```diff
@@ -347,26 +358,71 @@ def generate(self, max=10, submit=True):
             Submit the jobs to the queue.
         """
 
-        to_run = self._generate_hook()[:max]
+        to_run = self._generate_hook(user=user)[:max]
 
         for tag in to_run:
             queue_job(self.job_script(tag), submit=submit)
 
-    def _generate_hook(self):
+    def _generate_hook(self, user=None):
         """Override to add custom behaviour when jobs are queued."""
-        return self.pending()
+        return self.status(user=user)["not_yet_submitted"]
 
     def pending(self, user=None):
         """Jobs available to run."""
+        warnings.warn(
+            "'pending' method is deprecated. Call 'status()['not_yet_submitted']'."
+        )
+        return self.status(user)["not_yet_submitted"]
 
-        waiting, running = self.queued(user)
-        not_pending = set(self.ls()) | set(waiting) | set(running)
-        pending = [job for job in self.available() if job not in not_pending]
+    def failed(self, user=None, time_sort=False) -> Dict[str, list]:
```

Comment:
If you're adding type annotations to a method (👍 👍), then I would do the parameters too. Also, I usually don't add the types to the docstrings so they're only added in one place. I think the docstring parsers understand that and user the annotations.

Reply:
If you don't add a type to the docstring how do you typically format it? ex:
```python
        crashed : dict
            tags associated with each category
```
just becomes
```python
        crashed :
            tags associated with each category
```
??

Reply:
I think that the convention is to drop the colon too:
```
        crashed
            tags associated with each category
```